lib/Transforms/MemoryBanking.cpp

```diff
@@ -409,6 +409,151 @@ struct BankAffineStorePattern
   DenseSet<Value> &oldMemRefVals;
 };
 
+struct BankMemrefCopyPattern : public OpRewritePattern<memref::CopyOp> {
+  BankMemrefCopyPattern(MLIRContext *ctx, uint64_t bankingFactor,
+                        unsigned bankingDimension,
+                        DenseMap<Value, SmallVector<Value>> &memoryToBanks)
+      : OpRewritePattern<memref::CopyOp>(ctx), bankingFactor(bankingFactor),
+        bankingDimension(bankingDimension), memoryToBanks(memoryToBanks) {}
+
+  LogicalResult matchAndRewrite(memref::CopyOp copyOp,
+                                PatternRewriter &rewriter) const override {
+    Value srcMem = copyOp.getSource();
+    Value dstMem = copyOp.getTarget();
+
+    bool srcIsBanked = memoryToBanks.count(srcMem) != 0;
+    bool dstIsBanked = memoryToBanks.count(dstMem) != 0;
```

Comment:
Again, a good habit here to avoid double lookups is

```cpp
auto srcIt = memoryToBanks.find(srcMem);
auto dstIt = memoryToBanks.find(dstMem);
bool srcIsBanked = srcIt != memoryToBanks.end();
bool dstIsBanked = dstIt != memoryToBanks.end();

...

auto& srcBanks = srcIt->second;
```